lib/application.js

```diff
@@ -301,18 +301,16 @@ app.enableAuth = function() {
     var Model = method.ctor;
     var modelInstance = ctx.instance;
 
-    var modelId = modelInstance && modelInstance.id ||
-      // replacement for deprecated req.param()
-      (req.params && req.params.id !== undefined ? req.params.id :
-       req.body && req.body.id !== undefined ? req.body.id :
-       req.query && req.query.id !== undefined ? req.query.id :
-       undefined);
+    var modelId = modelInstance && modelInstance.id || ctx.args.id;
```

Comment:
I don't remember how exactly does strong-remoting handle arguments of prototype method. I think `ctx.args.id` is set for the shared constructor only, the method itself does not necessarily have to have an `id` argument.